src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTip.cs

```diff
@@ -0,0 +1,2565 @@
+					{
+						bool setFocus = CppWinRTHelpers.SetFocus(previouslyFocusedElement, FocusState.Programmatic);
+						m_previouslyFocusedElement = null;
+						args.Handled = setFocus;
+					}
+				}
+				else
+				{
+					Button GetF6Button()
+					{
+						var firstButton = m_closeButton;
+						var secondButton = m_alternateCloseButton;
+						//Prefer the close button to the alternate, except when there is no content.
+						if (CloseButtonContent == null)
+						{
+							(secondButton, firstButton) = (firstButton, secondButton);
+						}
+						if (firstButton != null && firstButton.Visibility == Visibility.Visible)
+						{
+							return firstButton;
+						}
+						else if (secondButton != null && secondButton.Visibility == Visibility.Visible)
+						{
+							return secondButton;
+						}
+						return null;
+					}
+
+					Button f6Button = GetF6Button();
+
+					if (f6Button != null)
+					{
+						f6Button.GettingFocus += (sender, args) =>
+						{
+							m_previouslyFocusedElement = new WeakReference<DependencyObject>(args.OldFocusedElement);
+						};
+						bool setFocus = f6Button.Focus(FocusState.Keyboard);
+						args.Handled = setFocus;
+					}
+				}
+			}
+		}
+
+		private void OnAutomationNameChanged(object sender, object args)
+		{
+			SetPopupAutomationProperties();
+		}
+
+		private void OnAutomationIdChanged(object sender, object args)
+		{
+			SetPopupAutomationProperties();
+		}
+
+		private void OnCloseButtonClicked(object sender, RoutedEventArgs args)
+		{
+			CloseButtonClick?.Invoke(this, null);
+			m_lastCloseReason = TeachingTipCloseReason.CloseButton;
+			IsOpen = false;
+		}
+
+		private void OnActionButtonClicked(object sender, RoutedEventArgs args)
+		{
+			ActionButtonClick?.Invoke(this, null);
+		}
+
+		private void OnPopupOpened(object sender, object args)
+		{
+			UIElement uiElement10 = this;
+			if (uiElement10 != null)
+			{
+				var xamlRoot = uiElement10.XamlRoot;
+				if (xamlRoot != null)
+				{
+					m_currentXamlRootSize = xamlRoot.Size;
+					m_xamlRoot = xamlRoot;
+					xamlRoot.Changed += XamlRootChanged;
+				}
+			}
+			else
+			{
+				var coreWindow = CoreWindow.GetForCurrentThread();
+				if (coreWindow != null)
+				{
+					coreWindow.SizeChanged += WindowSizeChanged;
+				}
+			}
+
+			// Expand animation requires UIElement9
+			if (this is UIElement && SharedHelpers.IsAnimationsEnabled())
+			{
+				StartExpandToOpen();
+			}
+			else
+			{
+				// We won't be playing an animation so we're immediately idle.
+				SetIsIdle(true);
+			}
+
+			var teachingTipPeer = FrameworkElementAutomationPeer.FromElement(this) as TeachingTipAutomationPeer;
+			if (teachingTipPeer != null)
+			{
+				string GetNotificationString()
+				{
+					string GetAppName()
+					{
+						try
+						{
+							var package = Package.Current;
+							if (package != null)
+							{
+								return package.DisplayName;
+							}
+						}
+						catch { }
```

Comment:
Codacy found an issue: [Handle the exception or explain in a comment why it can be ignored.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6388179)